### DISCUSSION TITLE
Integration with libSDL

### DISCUSSION THREAD

**Discussion by ckkashyap** [libuv/libuv#4195] [Ideas]
What I love about libUV is that it only depends on C - which makes me trust that libUV folks care about portability :). I think that libSDL also very much aligned philosophically with libUV. I am working on a portable version of picolisp (that  depends only on a C compiler). I am currently working on libuv integration with picolisp here - https://github.com/lispware/minilisp/tree/libuv . I'd like to add libSDL support to it too. I was wondering if there are projects out there that have already managed to make libUV and libSDL play together.

My ultimate goal is to get portable graphics and audio - I'd love for some pointers.

**Comment by bnoordhuis** [comment]
There are several ways to skin that cat, it really depends on what you want to do. Do sdl and libuv run on the same thread and if so, who is in control?

**Reply by ckkashyap** [reply]
After I posted the message above, I've been considering the idea of using separate processes for sdl and libuv and using pipes for communication between them. I think that one way I can specify my requirement is to have a sample.c that opens up a window to take in keyboard/mouse inputs (using SDL) and sends them over a TCP connection (using libuv).
